source/agora/consensus/EnrollmentManager.d

```diff
@@ -423,6 +424,37 @@ public class EnrollmentManager
         return false;
     }
 
+    /***************************************************************************
+
+        Get the unregistered enrollments in the block
+        And this is arranged in ascending order with the utxo_key
+
+        Params:
+            enrolls = will contain the unregistered enrollments data if found
+
+        Returns:
+            Return true if the unregistered enrollment data was found
+
+    ***************************************************************************/
+
+    public bool getUnregisteredEnrollments (ref Enrollment[] enrolls) @trusted
+    {
+        import std.stdio;
+
+        enrolls.length = 0;
+        assumeSafeAppend(enrolls);
+        auto results = this.db.execute("SELECT val FROM validator_set" ~
+            " WHERE enrolled_height is null ORDER BY hex(key) ASC");
+
+        foreach (row; results)
+            enrolls ~= deserializeFull!Enrollment(row.peek!(ubyte[])(0));
+
+        if (enrolls.length > 0)
```

Comment:
Just `return enrolls.length > 0` will do. :)

Reply:
Actually..

@Geod24 I feel like a function should not have a success/failure return code if you can already determine at the call site if the `enrolls` array is empty or not. What do you think?

Reply:
Yeah it should probably just `return ref`.

Reply:
I changed the return type.